File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.ts
function characterReplacement(s: string, k: number): number {
    const freq = []
    let ans = 0
    let max = 0
    let l = 0

    for (let r = 0; r < s.length; r++) {
        freq[s[r]] = (freq[s[r]] || 0) + 1
        max = Math.max(max, freq[s[r]])

        if ((r-l+1-max) > k) {
            freq[s[l]]--
            l++
        }
        ans = Math.max(ans, r - l + 1)
    }

    return ans
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left = 0
        d = {}
        most = 0
        ans = 0

        for right in range(len(s)):
            char = s[right]
            d[char] = d.get(char, 0) + 1
            most = max(most, d[char])

            if (right+1-left-most) > k:
                d[s[left]] -= 1
                left += 1
            
            ans = max(ans, right+1-left)

        return ans

            

        